src/lib.rs

```diff
@@ -142,8 +198,16 @@ pub fn children(i: usize) -> Option<(usize, usize)> {
 }
 
 /// Returns only the left child of a node, with a depth
+///
+/// ## Examples
+/// ```rust
+/// assert_eq!(flat_tree::left_child_with_depth(0, 0), None);
+/// assert_eq!(flat_tree::left_child_with_depth(1, 1), Some(0));
+/// assert_eq!(flat_tree::left_child_with_depth(3, 2), Some(1));
+/// ```
 // TODO: handle errors
```

Comment:
@yoshuawuyts What kind of errors should be handled here?

Reply:
Oh yeah, good one! -- I believe there's a possibility to overflow if calling it on the left-most child in the tree. The question is whether we should panic of return a `Result` there. I believe my comment was placed with the intent of refactoring it to a `Result`. But I'm not sure anymore what the best approach is.

Reply:
Oh yeah, whoops. I think I meant `Option` here too. I agree it's probably good to have that here (: